drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/producer/ScenarioSimulationKogitoProducer.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.kogito.client.producer;
+
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Specializes;
+import javax.inject.Inject;
+
+import org.drools.workbench.screens.scenariosimulation.client.producers.ScenarioSimulationProducer;
+import org.drools.workbench.screens.scenariosimulation.kogito.client.commands.ScenarioSimulationKogitoStateControlManager;
+
+
+@Specializes
```

Comment:
Why this?
If now there is a command manager that works in Kogito only we should use interface to decouple and then provide proper bean based on environment.
Wdyt?

Reply:
@danielezonca That was my first implementation (https://github.com/kiegroup/drools-wb/pull/1390/commits/a93ac60420e379262f5347102944a78179a3895e). What I don't like of that solution, is that it will result in a empty bean in case of BusinessCentral. The scope of this change is to define a bean which manages `StateControlAPI` mechanism, which is present in Kogito environment only. Therefore, I think that kind of solution should be used when either BusinessCentral and Kogito define a logic for that manager. In this case I'm sure this manager is required on Kogito only, so I want to define it in Kogito scope only.

Reply:
@danielezonca Agreed changes applied, thank you!